opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3/v3_2_0/TASK_5964/UpdateClinicalStudyConfiguration.java

```diff
@@ -0,0 +1,68 @@
+package org.opencb.opencga.app.migrations.v3.v3_2_0.TASK_5964;
+
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.UpdateOneModel;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.opencb.opencga.catalog.db.mongodb.MongoDBAdaptor;
+import org.opencb.opencga.catalog.db.mongodb.OrganizationMongoDBAdaptorFactory;
+import org.opencb.opencga.catalog.migration.Migration;
+import org.opencb.opencga.catalog.migration.MigrationRuntimeException;
+import org.opencb.opencga.catalog.migration.MigrationTool;
+import org.opencb.opencga.core.models.study.configuration.ClinicalAnalysisStudyConfiguration;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Migration(id = "update_clinical_study_configuration",
+        description = "Setting new default Clinical Study Configuration status values, #TASK-5964",
+        version = "3.2.0",
+        language = Migration.MigrationLanguage.JAVA, domain = Migration.MigrationDomain.CATALOG, date = 20240610)
+public class UpdateClinicalStudyConfiguration extends MigrationTool {
+
+    @Override
+    protected void run() throws Exception {
+        ClinicalAnalysisStudyConfiguration clinicalAnalysisStudyConfiguration = ClinicalAnalysisStudyConfiguration.defaultConfiguration();
+        Document clinicalConfigurationDocument = convertToDocument(clinicalAnalysisStudyConfiguration);
+
+        Bson query = new Document();
+        Bson projection = Projections.include("internal.configuration.clinical", "fqn");
+
+        for (String collection : Arrays.asList(OrganizationMongoDBAdaptorFactory.STUDY_COLLECTION, OrganizationMongoDBAdaptorFactory.DELETED_STUDY_COLLECTION)) {
+            migrateCollection(collection, query, projection,
+                    (document, bulk) -> {
+                        Document internal = document.get("internal", Document.class);
+                        if (internal == null) {
+                            throw new MigrationRuntimeException("'internal' field not found in study '" + document.get("fqn") + "'");
```

Comment:
These `== null` checks are super ultra mega unlikely, right? Like a very unexpected status, right?

Reply:
Highly highly highly unlikely. That configuration ALWAYS exist so the only possibility for not getting that information would be because we have a catastrophic bug (in the migration because we haven't included those fields which I did or because the studies don't have that field which would be extremely worrying). I just need to add conditions, don't panic !!!